Assignments bug

peter-budo opened the pull request:
- Fixed bug with ghost1 assignment when expansion icon couldn't been  recognised. Now instead it will show empty 1px image (therefore no visible expansion icon)
- Removed assignments image map from adapter, and reused the other from resources (need it fix on single player mission stringIDs)
- Moved out expansion icons image map and made sure that empty image showed if not found in the map
- Added colour overlay when assignment is not completed on 100%
- Added colour overlay to medals and ribbons
- Changed overlaying colour tint to lighter

peter-budo committed:
- Fixed bug with ghost1 assignment when expansion icon couldn't been recognized. Now instead it will show empty 1px image (therefore no visible expansion icon)
- Removed assignments image map from adapter, and reused the other from resources (need it fix on single player mission stringIDs)
- Moved out expansion icons image map and made sure that empty image showed if not found in the map

peter-budo committed:
Added colour overlay when assignment is not completed on 100%

peter-budo committed:
- Added colour overlay to medals and ribbons
- Changed overlaying colour tint to lighter

peter-budo commented:
@karllindmark have look when you have time. Feel free to leave comments and I will be online later in the evening if there is any urgent matter you want to discuss

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/assignments/AssignmentImageMap.java:
```diff
@@ -55,12 +55,12 @@
             put("WARSAW_ID_P_AWARD_ASGN_NAME_45", R.drawable.as45);
 
             /*Single player assignments*/
```
Will this break the feed when it displays singleplayer acheivements?

peter-budo replied:
To be fair I do not know, because I though it was just some mistake when map originally created.

karllindmark commented on core/src/main/java/com/ninetwozero/bf4intel/json/assignments/AssignmentAward.java:
```diff
@@ -6,6 +6,8 @@
     @SerializedName("code")
     private String assignmentKey;
+    @SerializedName("stringID")
+    private String stringID;
```
We should probably call this "name", for better consistency :-)

peter-budo replied:
That is very general name, but `uniqueName` can be better

karllindmark commented:
I played around a bit with the overlay, and thought of an alternative solution to it:

<img src="http://puu.sh/5OE4m.png" alt="Image" />

This needs the following alterations:

Remove this element in item_assigment.xml:

```
<RelativeLayout
    android:id="@+id/colour_overlay"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:background="@color/item_locked_overlay"
/>
```

Change this code in AssignmentAdapter:

```
if(assignment.getCompletion() == 100) {
    view.findViewById(R.id.colour_overlay).setVisibility(View.INVISIBLE);
} else {
    view.findViewById(R.id.colour_overlay).setVisibility(View.VISIBLE);
}
```

... to this:

```
if (assignment.getCompletion() == 100) {
    view.setAlpha(1f);
} else {
    view.setAlpha(0.6f);
}
```

Just a little experiment I felt we could try (not sure if better or worse)!

peter-budo committed:
Fixes per github requests

- Replaced FrameLayout with setAlpha()
- Change name of field from stringID to uniqueName

karllindmark commented:
:+1:
